aten/src/ATen/native/native_functions.yaml

```diff
@@ -8421,6 +8421,43 @@
 - func: special_erfinv.out(Tensor self, *, Tensor(a!) out) -> Tensor(a!)
   python_module: special
 
+- func: special_xlog1py(Tensor self, Tensor other) -> Tensor
+  python_module: special
+  variants: function
+  structured_delegate: special_xlog1py.out
+
+- func: special_xlog1py.self_scalar(Scalar self, Tensor other) -> Tensor
+  python_module: special
+  variants: function
+  dispatch:
+    CPU, CUDA: special_xlog1py
+
+- func: special_xlog1py.other_scalar(Tensor self, Scalar other) -> Tensor
+  python_module: special
+  variants: function
+  dispatch:
+    CPU, CUDA: special_xlog1py
+
+- func: special_xlog1py.out(Tensor self, Tensor other, *, Tensor(a!) out) -> Tensor(a!)
+  structured: True
+  structured_inherits: TensorIteratorBase
+  python_module: special
+  variants: function
+  dispatch:
+    CPU, CUDA: special_xlog1py_out
+
+- func: special_xlog1py.self_scalar_out(Scalar self, Tensor other, *, Tensor(a!) out) -> Tensor(a!)
+  python_module: special
+  variants: function
+  dispatch:
+    CPU, CUDA: special_xlog1py_out
```

Comment:
Any reason these variants couldn't be structured too?

Reply:
Oh, it looks like these are all composites. In that case you probably don't want to specify CPU, CUDA dispatch on these

Reply:
ExplicitAutograd implies that you have to write derivative formulas separately for each variant.

Reply:
We are doing that if I m correct,
https://github.com/pytorch/pytorch/blob/6de11c2db37d2b64debb97af40147de1581f718a/tools/autograd/derivatives.yaml#L691-L699